UnityProject/Assets/Scripts/Core/Lighting/LightsHolder.cs

```diff
@@ -0,0 +1,170 @@
+		{
+			if (Lights.Contains(data) == false)
+			{
+				Logger.LogError("Could not find correct light source to remove.");
+				ClearHeldLights();
+				UpdateLights();
+				return;
+			}
+
+			//this gets called on the server
+			RemoveLightObject(data.Id);
+
+			Lights.Remove(data);
+			netIdentity.isDirty = true;
+		}
+
+		private void RemoveLightObject(int id)
+		{
+			foreach (var sprite in lightsParent.GetComponentsInChildren<LightSprite>())
+			{
+				if (sprite.GivenID == id) Despawn.ClientSingle(sprite.gameObject);
+			}
+		}
+
+		private void OnLightsListChange(SyncList<LightData>.Operation op, int index, LightData oldItem,
+			LightData newItem)
+		{
+			if (op == SyncList<LightData>.Operation.OP_CLEAR)
+			{
+				ClearHeldLights();
+			}
+			else if (op == SyncList<LightData>.Operation.OP_REMOVEAT)
+			{
+				//this gets called on the client
+				RemoveLightObject(oldItem.Id);
+			}
+			else
+			{
+				UpdateLights();
+			}
+		}
+
+		public void UpdateLights()
+		{
+			for (int i = 0; i < Lights.Count; i++)
+			{
+				if (lightsParent.childCount <= i || lightsParent.GetChild(i) == null)
+				{
+					var newLight = Spawn.ClientPrefab(lightSpriteObject, parent: lightsParent);
+					// Add the LightSprite component to the new game object
+					LightSprite lightSprite = newLight.GameObject.GetComponent<LightSprite>();
+					// Set the properties of the LightSprite component based on the corresponding data in the Lights list
+					SetLightData(Lights[i], lightSprite);
+					lightSprite.transform.localPosition = Vector3.zero;
+					// Set the correct facing direction so that the light sprite doesn't look in a different direction when it gets added to the player.
+					lightSprite.transform.rotation = Quaternion.Euler(0f, 0f, (float)currentOrientation * 90 + 90);
+				}
+				else
+				{
+					LightSprite lightSprite = lightsParent.GetChild(i).gameObject.GetComponent<LightSprite>();
+					SetLightData(Lights[i], lightSprite);
+				}
+			}
+		}
+
+		private void SetLightData(LightData data, LightSprite lightSprite)
+		{
+			lightSprite.Color = data.lightColor;
+			lightSprite.Shape = data.lightShape;
+			lightSprite.Sprite = data.lightSpriteObject.GetComponent<ItemLightControl>().ObjectLightSprite.Sprite;
+			lightSprite.transform.localScale = new Vector3(data.size, data.size, data.size);
+			if (data.Id != 0)
+			{
+				lightSprite.GivenID = data.Id;
+				return;
+			}
+			Logger.LogWarning("No id was given to lightSprite, assigning random one.", Category.Lighting);
+			var newId = Guid.NewGuid().GetHashCode();
+			data.Id = newId;
+			lightSprite.GivenID = newId;
+		}
+
+		private void ClearHeldLights()
+		{
+			foreach (var sprite in lightsParent.GetComponentsInChildren<LightSprite>())
+			{
+				Despawn.ClientSingle(sprite.gameObject);
+			}
+		}
+	}
+
+	public struct LightData
+	{
+		public int Id;
+		public Color lightColor;
+		public Light2D.LightSprite.LightShape lightShape;
+		public GameObject lightSpriteObject;
```

Comment:
by the way it will try networking this

Reply:
wdym?
I am trying to network it, whats the issue?

Reply:
you can't network game objects

Reply:
https://mirror-networking.gitbook.io/docs/manual/guides/data-types